src/fem/nrf_fem_protocol_api.h

```diff
@@ -0,0 +1,275 @@
+ * The protocol must then provide PA and LNA protocol configuration parameters before the functionality can be used by the protocol.
+ *
+ * When the PA/LNA module is configured, the stack can call the provided enable functions before radio activity
+ * to enable the PA or LNA timer configurations for the upcoming radio activity.
+ * The default behavior is automatic deactivation of PA/LNA on the radio DISABLED event.
+ * This can be disabled, so that a manual deactivation can be performed instead.
+ */
+
+#ifndef NRF_FEM_PROTOCOL_API_H__
+#define NRF_FEM_PROTOCOL_API_H__
+
+#include "nrf_fem_control_config.h"
+#include "nrf_fem_protocol_legacy_api.h"
+
+#include "nrf_ppi.h"
+#include "nrf_timer.h"
+
+typedef enum
+{
+    NRF_802154_FAL_PA  = 1 << 0,
+    NRF_802154_FAL_LNA = 1 << 1,
+    NRF_802154_FAL_ALL = NRF_802154_FAL_PA | NRF_802154_FAL_LNA
+} nrf_fal_functionality_t;
+
+/**
+ * @brief PA and LNA activation event types.
+ */
+typedef enum
+{
+    NRF_802154_FAL_EVENT_TYPE_TIMER,
+    NRF_802154_FAL_EVENT_TYPE_GENERIC,
+    NRF_802154_FAL_EVENT_TYPE_PPI,
+} nrf_802154_fal_event_type_t;
+
+/**
+ * @brief Frontend Abstraction Layer event.
+ *
+ * The event can be a Timer Compare event or a Any event.
+ * Register value is only used for Timer Compare event and should contain timer value to set the compare relative to.
+ */
+typedef struct
+{
+    union
+    {
+        struct
+        {
+            NRF_TIMER_Type * p_timer_instance;     /* Pointer to a 1 us resolution timer instance. */
+            uint32_t         counter_value;        /* Timer value when radio activity starts. */
+            uint8_t          compare_channel_mask; /* Mask of compare channels which may be used by the FEM to schedule its own tasks. */
+        } timer;
+        struct
+        {
+            uint32_t register_address; /* Address of event register. */
+        } generic;
+        struct
+        {
+            uint8_t ch_id;                    /* Number of the PPI which was provided. */
+        } ppi;
+    }                           event;
+    bool                        override_ppi; /* False to ignore PPI channel below and use the one set by application. True to use the PPI channel below. */
+    uint8_t                     ppi_ch_id;    /* PPI channel to be used for this event. */
+    nrf_802154_fal_event_type_t type;         /* Type of event source. */
+} nrf_802154_fal_event_t;
+
+#if ENABLE_FEM
+
+/**
+ * @brief Set up PA using the provided events for the upcoming radio transmission.
+ *
+ * Multiple configurations may be provided by repeating calls to this function (i.e you can set the activate and deactivate event in multiple calls,
+ * and the configuration is preserved between calls).
+ *
+ * If a timer event is provided, i.e. NRF_802154_PA_LNA_EVENT_TYPE_TIMER, the PA will be configured to (de)activate at the application-configured time gap
```

Comment:
```suggestion
 * If a NRF_802154_PA_LNA_EVENT_TYPE_TIMER timer event is provided, the PA will be configured to activate or deactivate at the application-configured time gap
```
`i.e.` or `e.g.`? Also, what is `time gap`?

Reply:
Here it's i.e. (or, 'that is').
Time gap is basically the period of time between the 'cause' and 'effect' events that we want to achieve.

Reply:
1) OK, use my suggestion then.
2) OK.